app/service/rest.py

```diff
@@ -140,15 +146,19 @@ def update_service(service_id):
     # Capture the status change here as Marshmallow changes this later
     service_going_live = fetched_service.restricted and not req_json.get('restricted', True)
 
-    if 'reply_to_email_address' in req_json:
-        create_or_update_email_reply_to(fetched_service.id, req_json['reply_to_email_address'])
-
     current_data = dict(service_schema.dump(fetched_service).data.items())
     current_data.update(request.get_json())
     update_dict = service_schema.load(current_data).data
+    dao_update_service(update_dict)
+
+    if 'reply_to_email_address' in req_json:
+        create_or_update_email_reply_to(fetched_service.id, req_json['reply_to_email_address'])
```

Comment:
Not sure how defensive we want to be but are we guaranteed that req_json['reply_to_email_address'] will contain a value i.e. would you want to pass in an empty string?

Reply:
As of https://github.com/alphagov/notifications-admin/pull/1511, we no longer allow `reply_to_email_address` to be passed with an empty value.

I don't think it's worth the change, simple because we'll be deprecating marshmallow at some point and this validation will be refactored/removed anyway to support adding multiple letter contact blocks.